sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -84,10 +143,10 @@ public static class SchemaDoesntMatchException extends SchemaConversionException
           .put("NUMERIC", Type.TYPE_STRING) // Pass through the JSON encoding.
           .put("BIGNUMERIC", Type.TYPE_STRING) // Pass through the JSON encoding.
           .put("GEOGRAPHY", Type.TYPE_STRING) // Pass through the JSON encoding.
-          .put("DATE", Type.TYPE_STRING) // Pass through the JSON encoding.
+          .put("DATE", Type.TYPE_INT32)
           .put("TIME", Type.TYPE_STRING) // Pass through the JSON encoding.
           .put("DATETIME", Type.TYPE_STRING) // Pass through the JSON encoding.
```

Comment:
TIME and DATETIME can be Int64 so that the encoding size is smaller. Shouldn't block this PR:
https://github.com/googleapis/java-bigquerystorage/blob/main/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/JsonToProtoMessage.java#L334

Reply:
done